bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,722 @@
+    }
+
+    private void sendCommand(String path) {
+        String url = buildHttpAddress(path);
+        String status = null;
+        try {
+            status = HttpUtil.executeUrl("GET", url, 1000);
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+        logger.debug("Status received: {}", status);
+    }
+
+    private void refreshFromState(ChannelUID channelUID) {
+        switch (channelUID.getId()) {
+            case CHANNEL_STATE:
+                updateState(channelUID, StringType.valueOf(currentState));
+                break;
+            case CHANNEL_CLEAN:
+                updateState(channelUID, currentCleanState);
+                break;
+            case CHANNEL_START:
+                updateState(channelUID, currentStartState);
+                break;
+            case CHANNEL_HOME:
+                updateState(channelUID, currentHomeState);
+                break;
+            case CHANNEL_BATTERY:
+                updateState(channelUID, currentBattery);
+                break;
+            case CHANNEL_CPU_LOAD:
+                updateState(channelUID, currentCPULoad);
+                break;
+            case CHANNEL_SRV_MEM:
+                updateState(channelUID, StringType.valueOf(currentSrvMem));
+                break;
+            case CHANNEL_TURBO:
+                updateState(channelUID, currentTurbo);
+                break;
+            case CHANNEL_REPEAT:
+                updateState(channelUID, currentRepeat);
+                break;
+            case CHANNEL_MODE:
+                updateState(channelUID, StringType.valueOf(currentMode));
+                break;
+            case CHANNEL_NICKNAME:
+                updateState(channelUID, StringType.valueOf(currentNickname));
+                break;
+            case CHANNEL_CAMERA:
+                parseImage();
+                updateState(channelUID, currentImage);
+                break;
+            case CHANNEL_LAST_CLEAN:
+                updateState(channelUID, currentLastClean);
+                break;
+            case CHANNEL_MAP:
+                parseMap();
+                updateState(channelUID, currentMap);
+                break;
+            case CHANNEL_MONDAY:
+            case CHANNEL_TUESDAY:
+            case CHANNEL_WEDNESDAY:
+            case CHANNEL_THURSDAY:
+            case CHANNEL_FRIDAY:
+            case CHANNEL_SATURDAY:
+            case CHANNEL_SUNDAY:
+                refreshCounter = 1;
+                break;
+            default:
+                logger.warn("Channel refresh for {} not implemented!", channelUID.getId());
+        }
+    }
+
+    private void updateAllChannels() {
+        if (disposed) {
+            return;
+        }
+        if (refreshCounter > 0) {
```

Comment:
Reading quickly it can either be 0 or 1, what is going on here? What does it mean? Please note that even an int is not thread-safe and we are in a multi-threaded environment here.